tests: bluetooth: Update ICS for Mesh v1.1, DFU v1.0 and MBT v1.0

m-alperen-sener opened the pull request:
Mesh v1.1, DFU v1.0 and MBT v1.0 are supported by default in the stack. In this regard, this commit updates the ICS file and Launch Studio Project file to reflect supported features. Also, features that are neither supported nor fully implemented are dropped from the selection.

MESH

Mesh v1.0.1 is dropped and Mesh v1.1 is now selected.

Added features and models:
- Provisioning algorithm ECDH_P256 + HMAC_SHA256 + CCM_AES128.
- Node Provisioning Protocol Interface.
- Node address and composition refresh procedures.
- Private beacon server and client models.
- Remote provisioning server and client models.
- Composition data page 1 and 2.
- On-demand proxy server and client models.
- Solicitation PDU RPL configuration server model.
- Sending Proxy Solicitation.
- SAR configuration server and client models.
- Opcodes aggregator server and client models.
- Large composition data server and client models.
- Advertising with private network and node identity.

MBT

Mesh binary large object transfer model is added. Both client and
server models are supported.

DFU

Mesh device firmware update models are added.
- Firmware update server with pull/push mode BLOB transfer.
- Firmware distribution server with pull/push mode BLOB
transfer, store OOB firmware procedure and multiple image
support.
- Firmware update client with BLOB transfer.

PavelVPV commented on tests/bluetooth/qualification/ICS_Zephyr_Bluetooth_Host.pts:
I guess git shows all lines as changed because the line ending is also changed from CR to CRLF. If you convert the eol to CR in your file, we will sett the actual change.

sjanc replied:
please leave it as it is generated by Launch Studio, otherwise it will not be possible to import it

sjanc commented:
This is not passing consistency check
DFU
11:C.2 | If [MESH] (7/4) is Supported then [DFU] (11/2) is Excluded
21:C.2 | If [MESH] (7/4) is Supported then [DFU] (21/2) is Excluded

m-alperen-sener commented:
For the Launch studio conflict please check https://bluetooth.atlassian.net/jira/software/c/projects/ES/issues/ES-24228

sjanc commented:
cool, I see it should be solved in next TCRL so LGTM!:)

PavelVPV commented:
Putting DNM as I'd like to see clarification on removal proxy client from testing (https://github.com/auto-pts/auto-pts/pull/1113#issuecomment-1944355772) before PR gets merged.

m-alperen-sener committed:
tests: bluetooth: Update ICS for Mesh v1.1, DFU v1.0 and MBT v1.0

Mesh v1.1, DFU v1.0 and MBT v1.0 are supported by default in the stack.
In this regard, this commit updates the ICS file and Launch Studio
Project file to reflect supported features.

Signed-off-by: Alperen Şener <alperen.sener@nordicsemi.no>

m-alperen-sener commented:
I updated the PR: not de-selecting any ICS that are previously selected.

PavelVPV commented:
Would you mind enabling 12.11 (Sending Proxy Solicitation PDU) since it is also implemented?

m-alperen-sener commented:
> Would you mind enabling 12.11 (Sending Proxy Solicitation PDU) since it is also implemented?

I don't know exactly if it is implemented, if you say so, let's enabled that also.

m-alperen-sener commented:
I needed to manually convert the line endings from DOS to UNIX: compliance check fails otherwise. And also the line endings are handled automatically by git depending on the OS you work.

Regarding this two particular files, Launch studio and PTS application; PTS does not care about line endings in .pts file so it is OK to keep, but Launch studio does care for .bls file. However it should be handled accordingly or we should find a way to avoid compliance check on this file.

sjanc commented:
last time I updated bls file,  PR was simply merged with failed style check..

m-alperen-sener commented:
> last time I updated bls file, PR was simply merged with failed style check..

Is this a CI issue, then? Or should we move on like this and wait LS to fix this issue, which is the actual solution IMO.

sjanc commented:
don't know, I consider this bls file a blob so imho style checks should not be run on it

carlescufi commented:
Compliance failure is just due to the fact that .bts (XML) files are to be uploaded as-is, and they do not comply with Zephyr's formatting standards.
